### PR TITLE
fix favorite servers with single quotes in name or desc

### LINKED ISSUE #290: [Bug] Servers with descriptions containing a single quote mark cannot be added to Favorites
**Have you read our FAQ/Wiki/#before-you-ask ?**
Yes

**Describe the bug**
Servers with descriptions containing a single quote mark (possibly more) cannot be added to Favorites.

**To Reproduce**
1. Host or find a server with a single quote mark in the description.
2. Click the server in the list, then click "Add to Favorites"
3. See server list refresh but server was not added to Favorites

**Expected behavior**
Any server should be able to be added to Favorites no matter what the description contains.

**Desktop (please complete the following information):**
 - OS + Version: W11, latest

**Additional context**
Unknown if a second single quote mark in a description, or other characters, change this observed behavior.

### PR DESCRIPTION
**Pull request by Trezub**
fixes #290

### CODE REVIEW

**Comment by KauamCosta** [comment]
Approve

File: ui/modules/multiplayer/multiplayer.js [hunk 1]
```diff
@@ -1167,7 +1167,7 @@ function removeFav(server) {
 }
 
 function saveFav() {
-	bngApi.engineLua('MPConfig.setFavorites(\'' + JSON.stringify(favorites) + '\')');
+	bngApi.engineLua(`MPConfig.setFavorites('${JSON.stringify(favorites).replace(/'/g, "\\'")}')`);
```
Comment (OfficialLambdax) [inline ui/modules/multiplayer/multiplayer.js:1170]: if you so choose to encode json you also have to decode it on the receiving end. that being in MPConfig.setFavorites()

**Review by OfficialLambdax — CHANGES_REQUESTED** [review]
This breaks the favorite feature entirely. The receiving end cannot process the encoded json.
fgdgdfgfdgdfgdffgd

**Comment by Starystars67** [comment]
Interesting that this is raised shortly after I already started working on this one :P

https://github.com/BeamMP/BeamMP/tree/fix-290

In fact looks to be a direct copy paste of the work I did.

**Pull request closed without merging**